bluepysnap/spike_report.py

```diff
@@ -93,33 +94,79 @@ def get(self, group=None, t_start=None, t_stop=None):
         """Fetch spikes from the report.
 
         Args:
-            group (int/list/np.array/dict): Get spikes filtered by group. See NodePopulation.
+            group (None/int/list/np.array/dict): Get spikes filtered by group. See NodePopulation.
             t_start (float): Include only spikes occurring after this time.
             t_stop (float): Include only spikes occurring before this time.
 
         Returns:
-            pandas.Series/np.array:
-                If single node ID is passed as ``group`` returns a np.array of spiking times.
-                Otherwise return spiking node_ids indexed by sorted spike time.
+            pandas.Series: return spiking node_ids indexed by sorted spike time.
         """
         node_ids = [] if group is None else self._resolve_nodes(group).tolist()
 
         t_start = -1 if t_start is None else t_start
         t_stop = -1 if t_stop is None else t_stop
-        series_name = "{}_node_ids".format(self._population_name)
-
+        series_name = "ids"
         res = self._spike_population.get(node_ids=node_ids, tstart=t_start, tstop=t_stop)
         if not res:
             return pd.Series(data=[], index=pd.Index([], name="times"), name=series_name)
 
         res = pd.DataFrame(data=res, columns=[series_name, "times"]).set_index("times")[series_name]
         if self._sorted_by != "by_time":
             res.sort_index(inplace=True)
-        if np.issubdtype(type(group), np.integer):
-            return res.index.to_numpy()
         return res
 
 
+class FilteredSpikeReport(object):
+    """Access to filtered SpikeReport data."""
+    def __init__(self, spike_report, group=None, t_start=None, t_stop=None):
+        """Initialize a FilteredSpikeReport.
+
+        A FilteredSpikeReport is a lazy and cached object which contains the filtered data
+        from all the populations of a report.
+
+        Args:
+            spike_report (SpikeReport): The SpikeReport to filter.
+            group (None/int/list/np.array/dict): Get spikes filtered by group. See NodePopulation.
+            t_start (float): Include only frames occurring after this time.
+            t_stop (float): Include only frames occurring before this time.
+
+        Returns:
+            FilteredSpikeReport: A FilteredFrameReport object.
+        """
+        self.spike_report = spike_report
+        self.group = group
+        self.t_start = t_start
+        self.t_stop = t_stop
+
+    @cached_property
+    def report(self):
+        """Access to the report data.
+
+        Returns:
+            pandas.DataFrame: A DataFrame containing the data from the report. Row's index is the
```

Comment:
plural versus singular issue: "Row's index is ... timestamps".

Reply:
done